### ISSUE TITLE
volodya - Side effects of LTV = 0 assets: Index's users will not be able to withdraw (collateral), borrow

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-05-Index-judging#81]
volodya

high

# Side effects of LTV = 0 assets: Index's users will not be able to withdraw (collateral), borrow

## Summary
[Link to report from spearbit](https://solodit.xyz/issues/16216) 
When an AToken has LTV = 0, Aave restricts the usage of some operations. In particular, if the user
owns at least one AToken as collateral that has LTV = 0, operations could revert.
1) Withdraw: if the asset withdrawn is collateral, the user is borrowing something, the operation will revert if the
withdrawn collateral is an AToken with LTV > 0.
2) Transfer: if the from is using the asset as collateral, is borrowing something and the asset transferred is an
AToken with LTV > 0 the operation will revert.
3) Set the reserve of an AToken as not collateral: if the AToken you are trying to set as non-collateral is an
AToken with LTV > 0 the operation will revert.
Note that all those checks are done on top of the "normal" checks that would usually prevent an operation, de-
pending on the operation itself 
## Vulnerability Detail
```solidity
    function _validateNewCollateralAsset(ISetToken _setToken, IERC20 _asset) internal view {
        require(!collateralAssetEnabled[_setToken][_asset], "Collateral already enabled");

        (address aToken, , ) = protocolDataProvider.getReserveTokensAddresses(address(_asset));
        require(address(underlyingToReserveTokens[_asset].aToken) == aToken, "Invalid aToken address");

        ( , , , , , bool usageAsCollateralEnabled, , , bool isActive, bool isFrozen) = protocolDataProvider.getReserveConfigurationData(address(_asset));
        // An active reserve is an alias for a valid reserve on Aave.
        // We are checking for the availability of the reserve directly on Aave rather than checking our internal `underlyingToReserveTokens` mappings,
        // because our mappings can be out-of-date if a new reserve is added to Aave
        require(isActive, "IAR");
        // A frozen reserve doesn't allow any new deposit, borrow or rate swap but allows repayments, liquidations and withdrawals
        require(!isFrozen, "FAR");
        require(usageAsCollateralEnabled, "CNE");
    }

```
[v1/AaveV3LeverageModule.sol#L1101](https://github.com/sherlock-audit/2023-05-Index/blob/main/index-protocol/contracts/protocol/modules/v1/AaveV3LeverageModule.sol#L1101)
## Impact
The Index protocol might stop working for users who would use those particular markets.
## Tool used

Manual Review

## Recommendation
Add restriction to those markets and add documentation to restrict those markets
E.x.

```diff
    function _validateNewCollateralAsset(ISetToken _setToken, IERC20 _asset) internal view {
        require(!collateralAssetEnabled[_setToken][_asset], "Collateral already enabled");

        (address aToken, , ) = protocolDataProvider.getReserveTokensAddresses(address(_asset));
        require(address(underlyingToReserveTokens[_asset].aToken) == aToken, "Invalid aToken address");

-        ( , , , , , bool usageAsCollateralEnabled, , , bool isActive, bool isFrozen) = protocolDataProvider.getReserveConfigurationData(address(_asset));
+        ( , uint256 ltv , , , , bool usageAsCollateralEnabled, , , bool isActive, bool isFrozen) = protocolDataProvider.getReserveConfigurationData(address(_asset));

        // An active reserve is an alias for a valid reserve on Aave.
        // We are checking for the availability of the reserve directly on Aave rather than checking our internal `underlyingToReserveTokens` mappings,
        // because our mappings can be out-of-date if a new reserve is added to Aave
        require(isActive, "IAR");
+      require(ltv != 0, "ltv should be non zero");

        // A frozen reserve doesn't allow any new deposit, borrow or rate swap but allows repayments, liquidations and withdrawals
        require(!isFrozen, "FAR");
        require(usageAsCollateralEnabled, "CNE");
    }

```

Duplicate of #159
